types.go

```diff
@@ -908,6 +935,37 @@ type (
 		NetAmount           PurchaseUnitAmount  `json:"net_amount"`
 		TotalRefundedAmount *PurchaseUnitAmount `json:"total_refunded_amount,omitempty"`
 	}
+
+	ReferralRequest struct {
+		TrackingID    string      `json:"tracking_id"`
+		Operations    []Operation `json:"operations,omitempty"`
+		Products      []string    `json:"products,omitempty"`
+		LegalConsents []Consent   `json:"legal_consents,omitempty"`
+	}
+
+	Operation struct {
+		Operation                string             `json:"operation"`
+		APIIntegrationPreference IntegrationDetails `json:"api_integration_preference,omitempty"`
```

Comment:
nit: pointer otherwise it doesnt omitempty

Reply:
This is required if Operation is `API_INTEGRATION`, but not otherwise, so I'll make it a pointer.